File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
    
    stack<int> s1;
    stack<int> s2;
    int peekValue;
    int size = 0;
public:
    MyQueue() {
        s1 = stack<int>();
        s2 = stack<int>();
    }
    
    void push(int x) {
        s1.push(x);
        if(size==0)
            peekValue = x;
        size++;
    }
    
    int pop() {
        for(int i=0;i<size-1;i++) {
            s2.push(s1.top());
            s1.pop();
        }
        if(size > 1)
            peekValue = s2.top();
        int result = s1.top();
        s1.pop();
        while(!s2.empty()){
            s1.push(s2.top());
            s2.pop();
        }
        size--;
        return result;
    }
    
    int peek() {
        return peekValue;
    }
    
    bool empty() {
        return size == 0;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {
    
    private Stack<Integer> stack;
    private Integer front = 0;

    public MyQueue() {
        stack = new Stack<>();
        front = 0;
    }
    
    public void push(int x) {
        if(stack.isEmpty())
            front = x;
        stack.push(x);
    }
    
    public int pop() {
        Stack<Integer> temp = new Stack<>();
        while(!stack.isEmpty())
            temp.push(stack.pop());
        int pop = temp.pop();
        if (!temp.isEmpty())
            front = temp.peek();
        while(!temp.isEmpty())
            stack.push(temp.pop());
        return pop;
    }
    
    public int peek() {
        return front;
    }
    
    public boolean empty() {
        return stack.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */